src/components/sync-connected-sites.tsx

```diff
@@ -54,13 +63,35 @@ export const convertTreeToOptionsToSync = ( tree: TreeNode[] ): SyncOption[] =>
 		const wpContent = filesAndFolders.find( ( node ) => node.id === 'wp-content' )?.children || [];
 
 		wpContent.forEach( ( item ) => {
-			if ( item.checked && isSyncOption( item.id ) ) {
+			if ( ! isSyncOption( item.id ) ) {
+				return;
+			}
+
+			if ( item.checked || item.indeterminate ) {
 				optionsToSync.push( item.id );
 			}
+
+			if (
+				item.children &&
+				( item.id === SYNC_OPTIONS.plugins || item.id === SYNC_OPTIONS.themes )
+			) {
+				const selectedItems = item.children
+					.filter( ( child ) => child.checked )
+					.map( ( child ) => child.id );
+				if ( selectedItems.length > 0 && selectedItems.length < item.children.length ) {
+					specificSelections = {
+						...specificSelections,
+						[ item.id ]: selectedItems,
+					};
+				}
+			}
```

Comment:
I have tested one more time nd still can't understand why do we need those extra condiotions:

1. If parent item(`plugins` | `themes`) is selected, then we don't need to check for children, it's redundant operation
2. If parent is not selected, then we shoudl check for any selected child.
```suggestion
			if ( item.checked ) {
				optionsToSync.push( item.id );
				return;
			}

			if (
				item.children &&
				( item.id === SYNC_OPTIONS.plugins || item.id === SYNC_OPTIONS.themes )
			) {
				const selectedItems = item.children
					.filter( ( child ) => child.checked )
					.map( ( child ) => child.id );
				if ( selectedItems.length > 0 ) {
					specificSelections = {
						...specificSelections,
						[ item.id ]: selectedItems,
					};
				}
			}
```

Reply:
We did huge investigation and:
1. `plugins` should be selected, even if we do granular sync, otherwise they will be ignored. So we should proceed with what we have now.
2. But we have a bug with selective sync and it exists even at production now - STU-587

So it's safe to merge this PR and proceed wioth this feature, but we should fix taht existing bug (or maybe agree that it's expected behaviour).